src/dataset/annotations_utils.py

```diff
@@ -69,3 +70,34 @@ def xywh_to_xyxy(bbox: List) -> List:
     y2 = y + h
 
     return [x, y, x2, y2]
+
+
+def really_agnostic_segmentation_nms(masks: List[np.ndarray], scores: List[float], threshold: float) -> List[int]:
+    """Perform a custom version of non-maximum suppression algorithm, where segmentation is the main criteria.
+    Also, instead of computing the IoU, we compare the area of the masks.
```

Comment:
What does "IoU" mean? I see you explain later, but meaning should appear on the first appearance of an acronym.

Reply:
```suggestion
    Also, instead of computing the Intersection over Union (IoU), we compare the area of the masks.
```